### PR TITLE
Add bit and slice read/write functionalities

### PR DESCRIPTION
**Pull request by johnjohnlin**

### CODE REVIEW

File: common/include/verilog_int.h [hunk 1]
```diff
@@ -555,29 +556,105 @@ struct vint {
...
+			v[msb_word_slice] &= msb_mask_slice;
+		}
 	}
 
-	template<unsigned num_bit_src, unsigned pos>
-	void SetSlice() const {
-		// TODO
+	template<unsigned lsb_pos, unsigned num_bit_slice>
+	void WriteSlice(const vint<false, num_bit_slice>& sl) {
+		constexpr unsigned msb_pos = lsb_pos + num_bit_slice - 1;
```
Comment (yodalee) [inline common/include/verilog_int.h:633]: should we static_assert on the comparison on `msb_pos` and `num_bit`?

File: common/include/verilog_int.h [hunk 2]
```diff
@@ -555,29 +556,105 @@ struct vint {
...
+		constexpr unsigned lsb_word_slice = lsb_pos / bw_word;
+		constexpr unsigned msb_word_slice = msb_pos / bw_word;
+		constexpr unsigned num_word_slice = detail::num_bit2num_word(num_bit_slice);
+		constexpr unsigned unused_lsb_slice = lsb_pos % bw_word;
+		if constexpr (unused_lsb_slice == 0) {
+			for (unsigned i = 0; i < num_word_slice; ++i) {
+				v[lsb_word_slice+i] |= sl.v[i];
```
Comment (yodalee) [inline common/include/verilog_int.h:640]: The |= seems a little bit violate the name `WriteSlice`. If the caller does not reset the bits, the value is `or` not `write`.
Perhaps make this function private so user cannot call it directly, force them to use SetSlice?

**Review by yodalee — APPROVED** [review]
Not obvious error to me.

File: common/include/verilog_int.h [hunk 3]
```diff
@@ -560,24 +560,56 @@ struct vint {
 
 	void SetBit(unsigned pos, bool value) const {
```
[the remark this answers is not in the document]
Comment (johnjohnlin) [reply common/include/verilog_int.h]: Fixed

File: common/include/verilog_int.h (hunk 2 above)
Comment (johnjohnlin) [reply common/include/verilog_int.h:642]: Actually I use it elsewhere, let me rename it.

**Pull request merged**